### PR TITLE
Style/standardize productuploadmodal design

### LINKED ISSUE #59: Standardize ProductUploadModal Design
## 📘 Description

The `ProductUploadModal` needs to adopt standardized **shadcn/ui** components for a consistent and responsive design. Additionally, the component will need to be renamed, and an element should be repositioned to improve organization and alignment with the overall style. These changes will help maintain a unified user interface and improve the code structure.

## 🔢 Steps

1. Replace the current elements in `ProductUploadModal` with **shadcn/ui** components.
2. Rename the component to `AddProductModal`.
3. Center the modal's title.
4. Rename the file to `add-product-modal.tsx`.
5. Move the file to the `components/marketplace` folder, creating the folder if it does not already exist.

## ✅ Acceptance Criteria

- The modal uses the **shadcn/ui** components.
- The component has been renamed to `AddProductModal`.
- The modal's title is centered.
- The file has been renamed and moved correctly.
- The modal design is responsive.

## 🌎 References

- [Input](https://ui.shadcn.com/docs/components/input)
- [Label](https://ui.shadcn.com/docs/components/label)
- [Select](https://ui.shadcn.com/docs/components/select)
- [Textarea](https://ui.shadcn.com/docs/components/textarea)

## 📜 Additional Notes

- The form does not need to be functional at this stage; focus on design only.

⚠ **Do not apply until the ODHack begins.**
⚠ **Read our guidelines before applying.**

### PR DESCRIPTION
**Pull request by salazarsebas**
# 📝 Pull Request Title
Standardize ProductUploadModal Design

## 🛠️ Issue
- Closes #59 

## 📖 Description
- This pull request adds the shadcn/ui components so that they can be used in the file as needed.
- The pull request also implements the shadcn/ui components in the file that was renamed as add-product-modal.

## ✅ Changes made
- The shadcn/ui components are added.
- Rename the file as add-product-modal.
- It adapts the shadcn/ui components to add-product-modal.

## 🖼️ Media (screenshots/videos)
- Video:
https://www.loom.com/share/7e49220be4e7434b89e907feebdea199?sid=a1de9267-941b-4329-89e7-b03522225b96

### CODE REVIEW

File: frontend/app/components/ui/add-product-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { Button } from "./button";
+import { Input } from "./input";
+import { Label } from "./label";
+import { Select, SelectTrigger, SelectValue, SelectContent, SelectItem } from "./select";
+import { Textarea } from "./textarea";
+
+interface ProductUploadModalProps {
```
Comment (derianrddev) [inline frontend/app/components/ui/add-product-modal.tsx]: Rename the interface to AddProductModalProps

File: frontend/app/components/ui/add-product-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const ProductUploadModal: React.FC<ProductUploadModalProps> = ({
```
Comment (derianrddev) [inline frontend/app/components/ui/add-product-modal.tsx]: Rename the component to AddProductModal and here `React.FC<ProductUploadModalProps>` change it to   `React.FC<AddProductModalProps>`

File: frontend/app/components/header/subheader/SubHeader.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { Button } from "@radix-ui/themes";
 import BreadcrumbNavigation from "../../ui/breadcrumb-navigation";
 import { CirclePlus } from "lucide-react";
 import { useState } from "react";
-import ProductUploadModal from "../../ui/product-upload-modal";
+import AddProdcutModal from "../../ui/add-product-modal";
```
Comment (derianrddev) [inline frontend/app/components/header/subheader/SubHeader.tsx]: Rename `AddProdcutModal` to `AddProductModal`

File: frontend/app/components/header/subheader/SubHeader.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ const SubHeader = ({ name }: SubHeaderProps) => {
         </Button>
       </div>
 
-      <ProductUploadModal
+      <AddProdcutModal
```
Comment (derianrddev) [inline frontend/app/components/header/subheader/SubHeader.tsx]: Rename `AddProdcutModal` to `AddProductModal`

File: frontend/app/components/ui/add-product-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          >
+            Save
+          </Button>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default ProductUploadModal;
```
Comment (derianrddev) [inline frontend/app/components/ui/add-product-modal.tsx]: Here also, rename to AddProductModal

**Review by derianrddev — CHANGES_REQUESTED** [review]
@salazarsebas  Good job! Just some minor changes.

**Comment by salazarsebas** [comment]
Thank you very much for the corrections, I have already made them, can you confirm if everything is ok now?

**Review by derianrddev — APPROVED** [review]
@salazarsebas It looks good to me, we just need @danielcdz's approval to merge it.

**Comment by salazarsebas** [comment]
> It looks good to me, we just need @danielcdz's approval to merge it.

Excellent, thank you very much. I remain attentive.

**Review by danielcdz — APPROVED** [review]
LGTM

**Comment by salazarsebas** [comment]
> LGTM

tysm 🫡

**Pull request merged**